Bump to .NET 6.0

RussKie opened the pull request:
.NET 5 is nearing EOL, and .NET 6 is LTS.

mstv commented:
+1
It seems to work locally including automated tests but again not on AppVeyor. Still `draft`? 
I have reverted AppVeyor back to use VS 2022 and have addressed a few nitpicks.

Since the installation of VS 2022 and .NET6, a new GH sign-in dialog popped up - affects the .NET5 builds, too.

RussKie commented:
Thank you @mstv!

One thing I struggle to explain or pinpoint the culprit is why UI tests now run much longer than on .NET 5:
[net5.0 build](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/42468965): "_... Duration: **2 m 53 s** - UI.IntegrationTests.dll (net5.0)_"
[net 6.0 build](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/42471732): "..._Duration: **6 m 5 s** - UI.IntegrationTests.dll (net6.0)_"

mstv commented:
The integration test `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false` hung when trying to open Browse for `C:\`.
The FormBrowse showed the Dashboard instead of "not a repo".
After opening one of the recent repos, the tests continued (but failed with the errors addressed / ignored by #9856).

This test creates an empty temp directory now (with many lines of code).
BTW: The function `Path.GetTempFileName()` of .NET6(?) sucks because it returns names of existing files.

And I have removed an obsolete(?) file from the installer:
```
C:\projects\gitextensions\Setup\Product.wxs(194): error LGHT0103: The system cannot find the file 'C:\projects\gitextensions\artifacts\Release\publish\GitExtensions\\stdole.dll'.
```

RussKie commented:
[This build](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/42471834) took 12 minutes: "..._Duration: **12 m** - UI.IntegrationTests.dll (net6.0)_"

RussKie commented:
> And I have removed an obsolete(?) file from the installer:
> 
> ```
> C:\projects\gitextensions\Setup\Product.wxs(194): error LGHT0103: The system cannot find the file 'C:\projects\gitextensions\artifacts\Release\publish\GitExtensions\\stdole.dll'.
> ```

👍

RussKie commented:
> BTW: The function `Path.GetTempFileName()` of .NET6(?) sucks because it returns names of existing files.

If you have a repro, please raise a bug in https://github.com/dotnet/runtime/.
I think I may have discovered a bug with XmlSerializer which necessitated the change to ResourceManager/Xliff/TranslationSerializer.cs, but I yet to test it in a standalone repo

mstv commented:
That testcase `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false` passed but took almost 9.5 minutes.

RussKie commented on ResourceManager/Xliff/TranslationSerializer.cs:
```diff
@@ -9,9 +9,13 @@ public static class TranslationSerializer
     {
         public static void Serialize(TranslationFile translation, string path)
         {
-            using TextWriter tw = new StreamWriter(path, false);
```
A workaround for a regression in .NET 6, see: https://github.com/dotnet/runtime/issues/64885

vbjay commented:
> > BTW: The function `Path.GetTempFileName()` of .NET6(?) sucks because it returns names of existing files.
> 
> If you have a repro, please raise a bug in https://github.com/dotnet/runtime/.
> I think I may have discovered a bug with XmlSerializer which necessitated the change to ResourceManager/Xliff/TranslationSerializer.cs, but I yet to test it in a standalone repo


Nope. It's done that since day 1

Creates a uniquely named, zero-byte temporary file on disk and returns the full path of that file.

mstv commented:
FormBrowse showing the Dashboard did not become idle not before I opened a repo via RDesktop:

```
22:22:28.9563908 switching to UI thread
22:22:28.9564016 waiting for idle
22:22:28.9564508 showing form
22:25:57.5378374 idle
FormBrowse available
22:25:57.7819115 form shown
```

It hangs in `UITests.RunForm<T>` at `await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();`.
CC: @sharwell

RussKie commented:
I think it's good to merge. Any objections?

RussKie commented:
I don't know why tests are slower, I had more time to investigate

On Tue, 29 Mar 2022, 7:27 am Gerhard Olsson, ***@***.***>
wrote:

> I think it's good to merge. Any objections?
>
> Any reason why build time is 13 instead of 9 minutes?
>
> I have not run this myself until now, no comments when running briefly.
> No review comments.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gitextensions/gitextensions/pull/9860#issuecomment-1081107630>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBTEXSQ7WYBFTLX3BLSMUDVCIIZVANCNFSM5NUUQZ6A>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

gerhardol commented:
Pushed an update that fixes build issue for me locally, it was already disabled in AppVeyor so it should not fix that issue (but 6 hanged builds was followed by 2 successes after this change).

Hanging in AppTitle tests
![image](https://user-images.githubusercontent.com/6248932/161343389-375bd09c-af96-4077-8f6d-e66e89ef562e.png)

RussKie commented:
> Hanging in AppTitle tests
> ![image](https://user-images.githubusercontent.com/6248932/161343389-375bd09c-af96-4077-8f6d-e66e89ef562e.png)

For me, it appears `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false` is causing the issue:
![image](https://user-images.githubusercontent.com/4403806/161409655-52f96dfe-a6d6-4e27-b9d4-8372c4e4c360.png)

This test expects the revision grid view, however it's "stuck" in the dashboard view. Looking further into it...

RussKie commented:
I don't believe this test is valid any longer - if the provided path doesn't contain a valid git repo we no longer show the revision grid and show the dashboard.

mstv commented:
> I don't believe this test is valid any longer - if the provided path doesn't contain a valid git repo we no longer show the revision grid and show the dashboard.

`RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false`
has not become invalid - but perhaps the test setup - because the RevisionGrid *can* have _no selection_.
E.g. create a new orphan branch (and do not commit to it yet) - works better with #9919 merged.

RussKie commented:
> `RunScript_with_arguments_with_s_option_with_RevisionGrid_without_selection_shall_display_error_and_return_false`
> has not become invalid - but perhaps the test setup - because the RevisionGrid _can_ have _no selection_.

Good point, reverted the removal and updated the test arrangement.

RussKie commented:
Looks like the build is back to ~9-10 min mark

RussKie commented:
Any further thoughts/objections? If not, I'm going to merge it in a day or two.

gerhardol commented:
> Any further thoughts/objections? If not, I'm going to merge it in a day or two.

Seem OK to me - we should ship a 4.0 alpha though....
